source/grid/grid_in.cc

```diff
@@ -688,10 +688,10 @@ void GridIn<dim, spacedim>::read_ucd (std::istream                            &i
 
       // we use an unsigned int because we
       // fill this variable through an read-in process
-      unsigned int material_id;
+      types::material_id new_material_id;
 
       in >> dummy          // cell number
-         >> material_id;
+         >> new_material_id;
```

Comment:
This is questionable. The question is what data type is used in the UCD format to represent material ids. This is independent of how we would like to *interpret* that information later on.

I suspect that UCD really does use an `int` here and that we will later want to cast that to a `types::material_id` value.

Reply:
This is not a change in function. We are using an `unsigned int` for `types::material_id` already.
If `UCD` really uses an `int` here, I totally agree.

Reply:
You misunderstand what I wanted to say (or more likely: I didn't explain it well): What is in the file format is what is in the file format. It doesn't depend on how exactly we define our `types::material_id` type, and so we shouldn't use our own type when we decide how we read the data from the file.

Reply:
Anyway, I just removed the commit from this PR.